tests/test_discovery.py

```diff
@@ -260,36 +263,36 @@ async def close_chans_before_nursery(
             get_reg = partial(aportal.run_from_ns, 'self', 'get_registry')
 
             async with tractor.open_nursery() as tn:
-                portal1 = await tn.run_in_actor(
-                    stream_forever,
-                    name='consumer1',
-                )
-                agen1 = await portal1.result()
-
-                portal2 = await tn.start_actor('consumer2', rpc_module_paths=[__name__])
-                agen2 = await portal2.run(stream_forever)
-
-                async with trio.open_nursery() as n:
-                    n.start_soon(streamer, agen1)
-                    n.start_soon(cancel, use_signal, .5)
-                    try:
-                        await streamer(agen2)
-                    finally:
-                        # Kill the root nursery thus resulting in
-                        # normal arbiter channel ops to fail during
-                        # teardown. It doesn't seem like this is
-                        # reliably triggered by an external SIGINT.
-                        # tractor.current_actor()._root_nursery.cancel_scope.cancel()
-
-                        # XXX: THIS IS THE KEY THING that happens
-                        # **before** exiting the actor nursery block
-
-                        # also kill off channels cuz why not
-                        await agen1.aclose()
-                        await agen2.aclose()
+                portal1 = await tn.start_actor(
+                    name='consumer1', enable_modules=[__name__])
+                portal2 = await tn.start_actor(
+                    'consumer2', enable_modules=[__name__])
+
+                async with (
```

Comment:
heh, yeah so turns out this style (tuple after the with) is not allowed pre-3.9 😢; though oddly [doesn't seem documented anywhere](https://docs.python.org/3/whatsnew/3.9.html)?

It's def failing [in ci](https://github.com/goodboy/tractor/runs/2459473341?check_suite_focus=true#step:5:35) where you can see there's no syntax error in 3.9.

So we'll have to delay this I guess until we do a 3.9+ pin.

Reply:
Made a wish list: https://github.com/goodboy/tractor/issues/207